File: packages/client-py/skyportal_py/summary_query.py
"""Typed endpoint functions for ``/api/summary_query``."""

from __future__ import annotations

import httpx

from skyportal_py._http import unwrap
from skyportal_py_models.summary_query import (
    SummaryQueryMatch as SummaryQueryMatch,
    SummaryQueryPost as SummaryQueryPost,
    SummaryQueryResults as SummaryQueryResults,
)


def post_summary_query(
    client: httpx.Client,
    payload: SummaryQueryPost,
) -> SummaryQueryResults:
    """Search for sources whose summaries match a query.

    The search runs against the vector store of source summaries, so it
    requires the server to be configured with an embeddings store and an
    OpenAI key (globally or in the requesting user's preferences).

    Parameters
    ----------
    client : httpx.Client
        Client from :func:`skyportal_py.create_client`.
    payload : SummaryQueryPost
        The query. Exactly one of ``q`` (a free-text query) and
        ``obj_id`` (find sources similar to that source's summary) must
        be given. ``k`` is the maximum number of sources to return and
        must satisfy ``1 <= k <= 100``; server default 5. ``z_min`` and
        ``z_max`` bound the redshift of the returned sources and
        ``classification_types`` restricts them to those
        classifications; omitting them applies no restriction.
    """
    response = client.post(
        "/api/summary_query",
        json=payload.model_dump(by_alias=True, exclude_none=True),
    )
    return SummaryQueryResults.model_validate(unwrap(response))

File: packages/client-ts/src/SummaryQuery.ts
/**
 * Typed endpoint functions for `/api/summary_query`.
 *
 * @since 1.0.0
 */

import {
    SummaryQueryResults,
    type SummaryQueryPost,
} from "skyportal-js-models/SummaryQuery";

import * as Http from "./Http.ts";

export * from "skyportal-js-models/SummaryQuery";

/**
 * Search for sources whose summaries match a query.
 *
 * The search runs against the vector store of source summaries, so it requires
 * the server to be configured with an embeddings store and an OpenAI key
 * (globally or in the requesting user's preferences).
 *
 * @since 1.0.0
 * @category Requests
 * @param payload - The query.
 */
export const postSummaryQuery = async (
    client: Http.Client,
    payload: SummaryQueryPost
): Promise<SummaryQueryResults> =>
    Http.decode(
        SummaryQueryResults,
        await Http.post(client, "/api/summary_query", Http.body(payload))
    );
